src/Carousel.js

```diff
@@ -151,7 +151,8 @@ class Carousel extends React.Component {
       let direction;
       if (
         (nextIndex === 0 && previousActiveIndex >= lastPossibleIndex) ||
-        previousActiveIndex <= nextIndex
+        (previousActiveIndex <= nextIndex &&
+          !(previousActiveIndex === 0 && nextIndex >= lastPossibleIndex))
```

Comment:
Not a bad fix, but still will have issues when children=2, no?

Reply:
If children=2, then lastPossibleIndex will always be equal to 1 and so would nextIndex.
Given that nextIndex is always 1, if previousActiveIndex is indeed zero or one which it would be, then the if condition would always be satisfied.
Thus, you are correct, however, I believe it's in a better condition this way as at least we fix the most common use cases.